spec/app/domain/conjur/fetch_trusted_proxies_spec.rb

```diff
@@ -0,0 +1,297 @@
+                                                                        .with(valid_host, {:account=> valid_account})
+                                                                        .and_return(nil)
+    allow(mocked_role_class_by_login_returns_nil_and_roleid_nil).to receive(:roleid_from_username)
+                                                                        .with(valid_account, valid_host)
+                                                                        .and_return(nil)
+
+    allow(mocked_role_class_by_login_returns_nil_and_roleid_throws_error).to receive(:by_login)
+                                                                                 .with(valid_host, {:account=> valid_account})
+                                                                                 .and_return(nil)
+    allow(mocked_role_class_by_login_returns_nil_and_roleid_throws_error).to receive(:roleid_from_username)
+                                                                                 .with(valid_account, valid_host)
+                                                                                 .and_raise("Error")
+
+    allow(mocked_role_class_list_no_duplications).to receive(:by_login)
+                                                         .with(valid_host, {:account=> valid_account})
+                                                         .and_return(mocked_role_list_no_duplications)
+    allow(mocked_role_list_no_duplications).to receive(:restricted_to)
+                                                   .and_return(trusted_proxies_list_no_duplications)
+
+    allow(mocked_role_class_list_duplications_1).to receive(:by_login)
+                                                        .with(valid_host, {:account=> valid_account})
+                                                        .and_return(mocked_role_list_duplications_1)
+    allow(mocked_role_list_duplications_1).to receive(:restricted_to)
+                                                  .and_return(trusted_proxies_list_duplications_1)
+
+    allow(mocked_role_class_list_duplications_2).to receive(:by_login)
+                                                        .with(valid_host, {:account=> valid_account})
+                                                        .and_return(mocked_role_list_duplications_2)
+    allow(mocked_role_list_duplications_2).to receive(:restricted_to)
+                                                  .and_return(trusted_proxies_list_duplications_2)
+
+    allow(mocked_account_validator).to receive(:call)
+                                           .with({:account=>valid_account})
+                                           .and_return(true)
+    allow(mocked_account_validator).to receive(:call)
+                                           .with({:account=>invalid_account})
+                                           .and_raise('ACCOUNT_NOT_EXIST_ERROR')
+  end
+  #  ____  _   _  ____    ____  ____  ___  ____  ___
```

Comment:
@eladkug Please review the following UTs, also see commit message, thanks :)

Reply:
ACCOUNT_DOES_NOT_EXIST_ERROR. or NO_ACCOUNT_ERROR